kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/filter/DenyCipherSuiteFilter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import io.netty.handler.ssl.CipherSuiteFilter;
+import io.netty.util.internal.EmptyArrays;
+
+/**
+ * This class has been built to filter cipher suites based on requirements for Kroxylicious
+ * If no ciphers are declared then the default platform ciphers will be used
+ * If ciphers are declared then they are used instead of the default platform ciphers
+ * Both lists would then have anything removed that wasn't supported or in the deniedCiphers
+ */
+public final class DenyCipherSuiteFilter implements CipherSuiteFilter {
+
+    public static final DenyCipherSuiteFilter INSTANCE = new DenyCipherSuiteFilter();
```

Comment:
The singleton idea doesn't really work.   There might be many VirtualClusters defined each with different cipher suite requirements.   Which one's rules would win?   They'd all call `io.kroxylicious.proxy.filter.DenyCipherSuiteFilter#deniedCiphers`.  It would be a race.

Get rid of the singleton and have let `VirtualCluster` construct an instance of ConfigurableCipherSuiteFilter, passing its own configuration.  Pass the constructed instance to the io.netty.handler.ssl.SslContextBuilder#ciphers(java.lang.Iterable<java.lang.String>, io.netty.handler.ssl.CipherSuiteFilter)

You probably want a ctor like this:

```java
ConfigurableCipherSuiteFilter(@Nullable List<String> allowed, @Nullable List<String> denied) {}

```

Reply:
Yes, good shout! I was using the 2 implemented Netty classes Identity* and Supported* as templates which I why I went down that route. Hadn't even thought. Will sort out